src/oscar/core/thumbnails.py

```diff
@@ -27,15 +31,18 @@ def generate_thumbnail(self, source, **opts):
         width, height = size.split('x')
         # Set `size` to `width` if `height` is not provided.
         size = size if height else width
-        return get_thumbnail(source, size, **opts)
+        thumbnail = get_thumbnail(source, size, **opts)
+        if not thumbnail:
+            logger.error('Failed to generate thumbnail for the image %s', source)
+        return thumbnail
 
     def delete_thumbnails(self, source):
         from sorl.thumbnail import delete
         from sorl.thumbnail.helpers import ThumbnailError
         try:
             delete(source)
-        except ThumbnailError:
-            pass
+        except ThumbnailError as e:
```

Comment:
Not sure if we need to log the error on delete? `ThumbnailError` would only be raised if `source` was falsy, in which case there is no image to delete in the first place?

Reply:
Yes, I re-checked the source and it looks like it will be raised only if the source is empty/corrupted.

Wanted to add logger here for the debug/informational purposes so that we won't just silently catch the exception.

We can decrease logger level to `warning` or lower. What do you think?